agent/base

```diff
@@ -16,11 +16,11 @@ if [[ -z "$pbench_run" ]]; then
         pbench_run=/var/lib/pbench-agent
     fi
     export pbench_run
-fi
-
-if [[ ! -d $pbench_run ]]; then
-    echo "[ERROR] pbench run directory, ${pbench_run}, does not exist" >&2
-    exit 1
+else
```

Comment:
What if we change the value of pbench_run` in the config file? Don't we wanna check then, whether the directory exists or not?

Reply:
Pbench will implicitly create the default run directory (from the config file or fallback default) if it doesn't already exist (`mkdir -p` at line 29 in `base` or `os.makedirs` at line 42 in `base.py`). However we make sure that only takes effect when the value of `pbench_run` does come from the config file; if it's specified by an environment variable, and doesn't exist, we fail.

That is, regardless of the value in the config file, we'll make sure it exists the first time we reference it. (And if something deletes it later, a subsequent tool will fail.) If the user overrides it with an environment variable, however, they're responsible for creating it first: we don't own it, so we make the existence check.

Reply:
Riya, `mkdir -p` at line 29 in `base` should take care of creating a new pbench_run directory if the value in config file is changed. However, if user overrides it with an environment variable as Dave mentioned they are responsible for the directory creation as well.

Reply:
Thanks